.github/workflows/check-data-model-directory-updates.yaml

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2024 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Check for changes to data_model directory without a sha update 
+
+on:
+  pull_request:
+    branches-ignore:
+      - 'dependabot/**'
```

Comment:
There is no reason dependabot should be allowed to update data-model. I think we should remove this restriction. It made sense for submodules (which is the purpose of dependaboty) however it does not make sense for data_model.